crml/cennzx/src/impls.rs

```diff
@@ -54,29 +86,61 @@ impl<T: Trait> BuyFeeAsset for Module<T> {
 	}
 }
 
+/// Provides an impl for the `SimpleAssetSystem` trait
+/// Used to integrate GA with CENNZX
+pub struct SimpleAssetShim<T: prml_generic_asset::Trait>(PhantomData<T>);
+
+impl<T: prml_generic_asset::Trait> SimpleAssetSystem for SimpleAssetShim<T> {
+	type AccountId = T::AccountId;
+	type AssetId = T::AssetId;
+	type Balance = T::Balance;
+	/// Transfer some `amount` of assets `from` one account `to` another
+	fn transfer(
+		asset_id: Self::AssetId,
+		from: &Self::AccountId,
+		to: &Self::AccountId,
+		amount: Self::Balance,
+	) -> DispatchResult {
+		// note: we don't emit a 'transferred' event with this method
+		prml_generic_asset::Module::<T>::make_transfer(asset_id, from, to, amount)
+	}
+	/// Get the liquid asset balance of `account`
+	fn free_balance(asset_id: Self::AssetId, account: &Self::AccountId) -> Self::Balance {
+		prml_generic_asset::Module::<T>::free_balance(asset_id, account)
+	}
+	/// Get the default asset/currency ID in the system
+	fn default_asset_id() -> Self::AssetId {
+		<prml_generic_asset::Module<T> as MultiCurrencyAccounting>::DefaultCurrencyId::asset_id()
+	}
+}
+
```

Comment:
I'm thinking runtime/src/impl.rs could be a better place for this implementation as we could then avoid coupling cennzx to generic asset internally.